routes/helpers/firewall.rb

```diff
@@ -2,7 +2,7 @@
 
 class Clover
   def firewall_list_dataset
-    @project.firewalls_dataset.authorized(current_account.id, "Firewall:view")
+    dataset_authorize(@project.firewalls_dataset, "Firewall:view")
```

Comment:
I actually liked using chaining. Wrapping the entire dataset with another method like `dataset_authorize` doesn't seem quite intuitive. Can't we somehow support `rodauth.session_value` improvement with chaining?

Reply:
Would could do something like `@project.firewalls_dataset.with_account_id(current_account_id).authorized("Firewall:view")`.

Would you prefer that approach?

Reply:
If we pass the account id as `.with_account_id(current_account_id)`, we can also keep it in the current signature `.authorized(current_account_id, "Firewall:view")`. I think the issue is that we can't access `current_account_id` (`rodauth.session_value`) in the Dataset module. Therefore, it appears we need to pass it for chaining regardless.

Reply:
Correct.  I don't think the `.with_account_id(current_account_id)` chaining approach will work with the planned changes to support personal access tokens transparently.  Maybe we could do `@project.firewalls_dataset.authorized(self, "Firewall:view")`, and have the Dataset#authorize code deal with the Roda scope?  That looks nicer, but feels like a layer violation to me.

Reply:
Yeah, it mixups the layers. We can revisit it in the future if the new method becomes bothering.